test/e2e/far_e2e_test.go

```diff
@@ -71,61 +95,57 @@ var _ = Describe("FAR E2e", func() {
 			if len(nodes.Items) < 1 {
 				Fail("there are no worker nodes in the cluster")
 			}
-			//TODO: Randomize the node selection & verify valid index
-			// run FA on the first worker node
-			nodeObj := nodes.Items[0]
-			testNodeName = nodeObj.Name
-
-			switch clusterPlatform.Status.PlatformStatus.Type {
-			case configv1.AWSPlatformType:
-				fenceAgent = fenceAgentAWS
-				nodeIdentifierPrefix = nodeIdentifierPrefixAWS
-				By("running fence_aws")
-			case configv1.BareMetalPlatformType:
-				fenceAgent = fenceAgentIPMI
-				nodeIdentifierPrefix = nodeIdentifierPrefixIPMI
-				By("running fence_ipmilan")
-			default:
-				Skip("FAR haven't been tested on this kind of cluster (non AWS or BareMetal)")
-			}
-
-			testShareParam, err := buildSharedParameters(clusterPlatform, fenceAgentAction)
-			if err != nil {
-				Fail("can't get shared information")
+			if secondRun {
```

Comment:
In general it would be good to keep tests independent from each other. You should be able to run 1 test without the others, or just invert the order.

Reply:
I agree.
just to be clear: it's fine to have a test that checks creation of two remeditions  - just do it in a single test (i.e create both remediation in that test scope)  instead of relying on another test to create the first one.

Reply:
IMHO two tests are fine, however the 2nd one should not depend on the 1st for indeed. Can be resolved by using a BeforeEach() for the 2nd test to switch nodes.